File: 0875-koko-eating-bananas/0875-koko-eating-bananas.js
/**
 * @param {number[]} piles
 * @param {number} h
 * @return {number}
 */
var minEatingSpeed = function(piles, h) {
    let min = 1, max = -Infinity
    for (const pile of piles) {
        max = Math.max(max, pile)
    }
    while (min < max) {
        const m = Math.floor((min+max)/2)
        if (canEat(m)) {
            max = m
        } else {
            min = m + 1
        }
    }
    return min
    
    function canEat(speed) {
        let hour = 0
        for (const pile of piles) {
            hour += Math.ceil(pile/speed)
        }
        return hour <= h
    }
};

File: 0875-koko-eating-bananas/0875-koko-eating-bananas.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        lo, hi = 1, max(piles)
        while lo < hi:
            m, hrs = (lo+hi)//2, 0
            for p in piles:
                hrs += ceil(p/m)
            if hrs <= h:
                hi = m
            else:
                lo = m + 1
        return lo